sdk/packages/selector-sui/src/adapter.ts

```diff
@@ -0,0 +1,495 @@
+    if (isMobileBrowser() && this._app) {
+      const mobileWalletName = getRecentStandardWalletForNetwork(SUI_NETWORK)
+      const wallet = this._walletsList.find((w) => w.name === mobileWalletName)
+
+      if (typeof wallet === 'undefined') {
+        return
+      }
+
+      if (wallet.deeplink === null) {
+        return
+      }
+
+      if (wallet.deeplink.universal !== null) {
+        this._app.connectDeeplink({
+          walletName: wallet.name,
+          url: wallet.deeplink.universal
+        })
+        return
+      }
+
+      if (wallet.deeplink.native !== null) {
+        this._app.connectDeeplink({
+          walletName: wallet.name,
+          url: wallet.deeplink.native
+        })
+      }
+    }
+  }
+
+  connectToMobileWallet = (walletName: string) => {
+    if (this._modal) {
+      this._modal.setStandardWalletConnectProgress(true)
+    }
+
+    const wallet = this._walletsList.find((w) => w.name === walletName)
+
+    if (!this._app || typeof wallet === 'undefined') {
+      return
+    }
+
+    if (wallet.deeplink === null) {
+      return
+    }
+
+    if (wallet.deeplink.universal !== null) {
+      this._app.connectDeeplink({
+        walletName: wallet.name,
+        url: wallet.deeplink.universal
+      })
+
+      this._chosenMobileWalletName = walletName
+
+      triggerConnect(
+        wallet.deeplink.universal,
+        this._app.sessionId,
+        this._appInitData.url ?? 'https://nc2.nightly.app'
+      )
+      return
+    }
+
+    if (wallet.deeplink.native !== null) {
+      this._app.connectDeeplink({
+        walletName: wallet.name,
+        url: wallet.deeplink.native
+      })
+
+      this._chosenMobileWalletName = walletName
+
+      triggerConnect(
+        wallet.deeplink.native,
+        this._app.sessionId,
+        this._appInitData.url ?? 'https://nc2.nightly.app'
+      )
+    }
+  }
+  connectToStandardWallet = async (walletName: string, onSuccess: () => void) => {
+    if (this._modal) {
+      this._modal.setStandardWalletConnectProgress(true)
+    }
+    const wallet = this._walletsList.find((w) => w.name === walletName)
+    if (typeof wallet?.standardWallet === 'undefined') {
+      return
+    }
+
+    const adapter = new StandardWalletAdapter({
+      wallet: wallet.standardWallet as StandardWalletAdapterWallet
+    })
+
+    try {
+      await adapter.connect()
+
+      persistRecentStandardWalletForNetwork(walletName, SUI_NETWORK)
+      persistStandardConnectForNetwork(SUI_NETWORK)
+      this._connectionType = ConnectionType.WalletStandard
+      this._innerStandardAdapter = adapter
+      this._accounts = (await adapter.getAccounts()).map((a) => a)
+      this.connected = true
+      this._connecting = false
+      this.connecting = false
+
+      this._modal?.closeModal()
+      onSuccess()
+    } catch (e) {
+      if (this._modal) {
+        this._modal.setStandardWalletConnectProgress(false)
+      }
+      clearRecentStandardWalletForNetwork(SUI_NETWORK)
```

Comment:
I don't think this should be done here, but only when newly connected wallet is nighty connect through qr code, like we do in solana adapter

Reply:
if you use this because adapter.getAccounts can throw errors, then it might be better to just move stuff from lines 459 and 460 after calling it

Reply:
So this is case that Sui selector auto connects and if we reject using wallet it will still try to do it on nightly connect rather than show connect modal.

Reply:
ok, shenanigans of sui selector, fine, leave it like it is